client/jmx-adapter/src/test/java/org/jolokia/client/jmxadapter/JmxBridgeTest.java

```diff
@@ -329,13 +329,14 @@ public void startAgent()
     // wait for agent to be running
     await()
         .until(
-            Awaitility.matches(
-                new ThrowingRunnable() {
-                  @Override
-                  public void run() throws J4pException {
-                    connector.execute(new J4pVersionRequest());
-                  }
-                }));
+        		new Callable<Boolean>() {
+
+					@Override
+					public Boolean call() throws Exception {
+						//will throw exception if connection is not ready
+						connector.execute(new J4pVersionRequest());
+						return true;
+					}});
```

Comment:
for less change, it is also possible to use:

```
    await()
        .untilAsserted(
                new ThrowingRunnable() {
                  @Override
                  public void run() throws J4pException {
                    connector.execute(new J4pVersionRequest());
                  }
                });
```

but well, it is one line less of code so both solutions seems fine